sonar-slang-plugin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java

```diff
@@ -192,11 +174,60 @@ private Tree createElement(@Nullable PsiElement element) {
       return new LiteralTreeImpl(metaData, element.getText());
     } else if (element instanceof KtOperationExpression) {
       return createOperationExpression((KtOperationExpression) element, metaData);
+    } else if (element instanceof KtParameter) {
+      return new IdentifierTreeImpl(metaData, ((KtParameter) element).getName());
     } else {
       return new NativeTreeImpl(metaData, new KotlinNativeKind(element), list(Arrays.stream(element.getChildren())));
     }
   }
 
+  @NotNull
+  private Tree createFunctionDeclarationTree(@NotNull KtFunction functionElement, TreeMetaData metaData) {
+    List<Tree> modifiers = getModifierList(functionElement.getModifierList());
+    PsiElement nameIdentifier = functionElement.getNameIdentifier();
+    Tree returnType = null;
+    IdentifierTree identifierTree = null;
+    List<Tree> parametersList = list(functionElement.getValueParameters().stream());
+    Tree bodyTree = createElement(functionElement.getBodyExpression());
+    KtTypeElement typeElement = functionElement.getTypeReference() != null ? functionElement.getTypeReference().getTypeElement() : null;
+
+    if (functionElement.hasDeclaredReturnType() && typeElement != null) {
+      returnType = new IdentifierTreeImpl(getTreeMetaData(typeElement), typeElement.getText());
+    }
+    if (nameIdentifier != null) {
+      identifierTree = new IdentifierTreeImpl(getTreeMetaData(nameIdentifier), functionElement.getName());
+    }
+    if (bodyTree != null) {
+      // FIXME are we sure we want body of function as block tree ?
+      if (!(bodyTree instanceof BlockTree)) {
+        bodyTree = new BlockTreeImpl(bodyTree.metaData(), Collections.singletonList(bodyTree));
+      }
+
+      // Set bodyTree to null for empty lambda functions
+      if (bodyTree.children().isEmpty()) {
+        bodyTree = null;
```

Comment:
It would be good to have a test to cover that case.

Reply:
Test added (bd329faf3c92cc56cc352e96e667a1ce9a5c06fa)